lib/matplotlib/axes/_base.py

```diff
@@ -968,50 +987,52 @@ def cla(self):
         self.callbacks = cbook.CallbackRegistry()
 
         if self._sharex is not None:
-            # major and minor are class instances with
-            # locator and formatter attributes
-            self.xaxis.major = self._sharex.xaxis.major
-            self.xaxis.minor = self._sharex.xaxis.minor
+            # The tickers need to exist but can be empty until after the
+            # call to Axis._set_scale since they will be overwritten
+            # anyway
+            self.xaxis.major = maxis.Ticker()
+            self.xaxis.minor = maxis.Ticker()
+
+            # Copy the axis limits
             x0, x1 = self._sharex.get_xlim()
             self.set_xlim(x0, x1, emit=False, auto=None)
 
-            # Save the current formatter/locator so we don't lose it
-            majf = self._sharex.xaxis.get_major_formatter()
-            minf = self._sharex.xaxis.get_minor_formatter()
-            majl = self._sharex.xaxis.get_major_locator()
-            minl = self._sharex.xaxis.get_minor_locator()
-
             # This overwrites the current formatter/locator
             self.xaxis._set_scale(self._sharex.xaxis.get_scale())
 
-            # Reset the formatter/locator
-            self.xaxis.set_major_formatter(majf)
-            self.xaxis.set_minor_formatter(minf)
-            self.xaxis.set_major_locator(majl)
-            self.xaxis.set_minor_locator(minl)
+            # Reset the formatter/locator. Axis handle gets marked as
+            # stale in previous line, no need to repeat.
+            if self._share_tickers:
```

Comment:
The old way of doing things was:
  1. Copy the `Ticker` objects to the new axis (lines 973-4)
  2. Save the formatters and locators (lines 979-982)
  3. Trash the formatters and locators with `_set_scale` (line 985)
  4. Restore the formatters and locators (lines 988-991)
This would restore for both the old and new axis since they always shared the same `Ticker` object.

I made this process a little more efficient:
  1. Set empty `Tickers` for the new axis (this one) (lines 993-4)
  2. Let `_set_scale` mess with the empty `Tickers` (line 1001)
  3. Reset the `Ticker`s to either a reference (lines 1006-7) or a copy (lines 1009-10) of the old axis's `Ticker`s.

Since `_share_tickers` defaults to `True`, the original behavior of having a reference to the original axis' `Ticker`s is preserved unless explicitly asked for by the user. The key here is that `_set_scale` messes up the formatters and locators for its own purposes and it's effects have to be undone one way or another. I think my way is just simpler.